File: hr_management_app/lib/models/employee_insurance.dart
class EmployeeInsurance {
  final int employeeId;
  final int insuranceId;
  final double actualEmployeePct;
  final double actualCompanyPct;
  final double actualTotalPct;
  final String? note;
  final String createdAt;
  final String updatedAt;

  EmployeeInsurance({
    required this.employeeId,
    required this.insuranceId,
    required this.actualEmployeePct,
    required this.actualCompanyPct,
    required this.actualTotalPct,
    this.note,
    required this.createdAt,
    required this.updatedAt,
  });

  factory EmployeeInsurance.fromJson(Map<String, dynamic> json) {
    return EmployeeInsurance(
      employeeId: json['employee_id'] ?? 0,
      insuranceId: json['insurance_id'] ?? 0,
      actualEmployeePct: (json['actual_employee_pct'] ?? 0.0).toDouble(),
      actualCompanyPct: (json['actual_company_pct'] ?? 0.0).toDouble(),
      actualTotalPct: (json['actual_total_pct'] ?? 0.0).toDouble(),
      note: json['note'],
      createdAt: json['created_at'] ?? '',
      updatedAt: json['updated_at'] ?? '',
    );
  }
}

File: backend/app/Models/EmployeeInsurance.php
<?php

namespace App\Models;

use Illuminate\Database\Eloquent\Factories\HasFactory;
use Illuminate\Database\Eloquent\Model;
use Illuminate\Database\Eloquent\Relations\BelongsTo;

class EmployeeInsurance extends Model
{
    use HasFactory;

    protected $table = 'employee_insurance';

    protected $fillable = [
        'employee_id',
        'insurance_id',
        'actual_employee_pct',
        'actual_company_pct',
        'actual_total_pct',
        'note',
    ];

    public function employee(): BelongsTo
    {
        return $this->belongsTo(Employee::class, 'employee_id');
    }

    public function insurance(): BelongsTo
    {
        return $this->belongsTo(Insurance::class, 'insurance_id');
    }
}